js/private/npm_import.bzl

```diff
@@ -234,15 +256,33 @@ _ATTRS = {
         location.""",
         default = ".",
     ),
+    "enable_lifecycle_hooks": attr.bool(
+        doc = """If true, runs any lifecycle hooks declared in this package.""",
+        default = True,
+    ),
     "experimental_reference_deps": attr.bool(
         doc = """Experimental reference deps allow dep to support circular deps between npm packages.
         This feature depends on dangling symlinks, however, which is still experimental in bazel,
-        has issues with "host" and "exec" configurations, and does not yet work with remote exection."""
-    ), 
+        has issues with "host" and "exec" configurations, and does not yet work with remote exection.""",
+    ),
 }
 
 npm_import = struct(
     doc = _DOC,
     implementation = _impl,
     attrs = _ATTRS,
 )
+
+def _has_lifecycle_hooks(rctx, package_path):
+    pkg_json_path = package_path + _sep(rctx) + "package.json"
+    pkg_json = json.decode(rctx.read(pkg_json_path))
+    return "scripts" in pkg_json and (
+        "preinstall" in pkg_json["scripts"] or
+        "install" in pkg_json["scripts"] or
+        "postinstall" in pkg_json["scripts"]
```

Comment:
thinking ahead to when we will add user postinstall steps to npm_import, my thinking was we'd use a `bazel_postinstall` of `rules_js_postinstall` script so we don't have to worry about appending to an existing postinstall. but... pnpm lifecycle hooks tooling won't run that at all. so I think we have two options,

1) find a way to get pnpm lifecycle tooling to also run our user postinstall script

2) if there is a user specified postinstall and an existing postinstall, then rename the existing one to "npm_postinstall" and add the "bazel_postinstall" and then insert a "postinstall" that runs both in some standard and platform agnostic way: `"postinstall": "npm run npm_postinstall && npm run bazel_postinstall"`?

Reply:
although, it might be that the `&&` in a`"postinstall": "existing commands && user commands"` is understood by pnpm lifecycle hooks and is already platform agnostic so lets try that first before overcomplicating

Reply:
can you add custom user postinstall commands via `translate_pnpm_lock` and `npm_import` to this diff to prove it out?